### PR TITLE
[release-v3] Temporary Fix for v3 Timeouts

### PR DESCRIPTION
**Pull request by dvonthenen**
This supersedes this PR:
https://github.com/deepgram/deepgram-dotnet-sdk/pull/179

I would have used that PR but there were Project Properties that needed to be updated beyond what was provided.

NOTE: This is going on the `release-v3` branch and will tag a release from there.

## Summary by CodeRabbit

- **Chores**
	- Updated project metadata including package ID, title, version, and authors.

- **Bug Fixes**
	- Implemented a temporary fix in the HTTP client utility to address a timeout control flow issue.

### CODE REVIEW

**Comment by dvonthenen** [comment]
@coderabbitai review

File: Deepgram/Utilities/HttpClientUtil.cs [hunk 1]
```diff
@@ -1,45 +1,45 @@
...
-                HttpClient = Create();
-            }
-
-            // Set the timeout
-            HttpClient.Timeout = timeSpan;
-        }
-    }
-}
+                // HttpClient = Create(); // temporary bug fix until SDK v4 is released
+            }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Utilities/HttpClientUtil.cs]: Removing per:
https://github.com/deepgram/deepgram-dotnet-sdk/issues/127#issuecomment-1779085524

This is the correct analysis.

File: Deepgram/Utilities/HttpClientUtil.cs [hunk 2]
```diff
@@ -1,45 +1,45 @@
...
+
+            // Set the timeout
+            HttpClient.Timeout = timeSpan;
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Utilities/HttpClientUtil.cs:35]: Removing per:
https://github.com/deepgram/deepgram-dotnet-sdk/issues/127#issuecomment-1779085524

This is the correct analysis.

File: Deepgram.Tests/UtilitiesTests/HttpClientUtilTests.cs [hunk 3]
```diff
@@ -21,7 +23,50 @@ public void GetUserAgent_Should_Return_HttpClient_With_Accept_And_UserAgent_Head
...
+                await client.GetAsync(endpoint);
+            }
+            catch (TaskCanceledException)
+            {
+                resultShouldSucceed = true;
+            }
 
+            // Assert
+            Assert.True(resultShouldFail);
+            Assert.False(resultShouldSucceed);
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram.Tests/UtilitiesTests/HttpClientUtilTests.cs]: Fixed spelling and addressed comments.

**Comment by dvonthenen** [comment]
Ok. So I have done a crap ton of analysis on this issue. The analysis here is correct but also has a ton of other consequences:
https://github.com/deepgram/deepgram-dotnet-sdk/issues/127#issuecomment-1779085524

Immediate problems - as the above analysis points out, we can't re-create the client because all of the clients have pointers to the initial httpclient created when the clients were initialized. we need to remove the creation of the client in order to set the timeout of the original clients. this also means that the timeout value is shared among all clients

things to note - a single `HttpClientUtil()` object is *not* thread-safe. Setting the timeout value to multiple different values in multiple threads based on need (for example, uploading a large file) might inadvertently increase the timeouts for other requests. The same is true for smaller values... you might inadvertently (like for a simple get) decrease the timeout for operations needing more time in another thread. the take away, you must have a `HttpClientUtil()` per thread.

**Review by lukeocodes — APPROVED** [review]

**Pull request merged**